lib/Service/AssistantService.php

```diff
@@ -36,14 +37,20 @@ public function sendNotification(Task $task, ?string $target = null, ?string $ac
 			'id' => $task->getId(),
 			'input' => $task->getInput(),
 			'target' => $target,
-			'actionLabel' => $actionLabel,
-			'taskTypeClass' => $task->getType(),
+			'actionLabel' => $actionLabel,			
 		];
+		if ($task instanceof TextToImageTask) {
+			$params['taskType'] = Application::TASK_TYPE_TEXT_TO_IMAGE;
+		} else {
+			$params['taskType'] = Application::TASK_TYPE_TEXT_GEN;
+			$params['textTaskTypeClass'] = $task->getType();
+		}
+
 		$status = $task->getStatus();
 		$subject = $status === Task::STATUS_SUCCESSFUL
 			? 'success'
 			: 'failure';
-		$objectType = ($task->getAppId() === Application::APP_ID || $target === null)
+		$objectType = ($task->getAppId() === Application::APP_ID && $target === null)
```

Comment:
I did this change since the text to image tasks need a different target from the text processing tasks but I guess we could also resolve that target in the notifier, as well. Only difference being that we also need to pass the task identifier to resolve the link.

Reply:
Note to self: $target === null is the correct condition here.

Reply:
:+1: